DigitalLearningSolutions.Data.Tests/Services/SessionServiceTests.cs

```diff
@@ -0,0 +1,166 @@
+
+                // Then
+                var newSession = sessionTestHelper.GetSession(sessionId);
+
+                newSession.Should().NotBeNull();
+                SessionTestHelper.SessionsShouldBeApproximatelyEquivalent(
+                    newSession!,
+                    new Session(
+                        sessionId,
+                        candidateId,
+                        customisationId,
+                        DateTime.Now,
+                        0,
+                        true
+                    )
+                );
+            }
+        }
+
+        [Test]
+        public void StartOrRestartSession_Should_Stop_Other_Sessions()
+        {
+            using (new TransactionScope())
+            {
+                // When
+                const int candidateId = 101;
+                const int customisationId = 1240;
+                var sessionId = sessionService.StartOrRestartSession(candidateId, customisationId);
+
+                // Then
+                var sessions = sessionTestHelper.GetCandidateSessions(candidateId);
+                var activeSessions = sessions.Where(session => session.Active);
+
+                activeSessions.Should().HaveCount(1);
+                activeSessions.First().SessionId.Should().Be(sessionId);
+            }
+        }
+
+        [Test]
+        public void StopSession_Should_Stop_Candidates_Sessions()
+        {
+            using (new TransactionScope())
+            {
+                // When
+                const int candidateId = 101;
+                sessionService.StopSession(candidateId);
+
+                // Then
+                var sessions = sessionTestHelper.GetCandidateSessions(candidateId);
+                var activeSessions = sessions.Where(session => session.Active);
+
+                activeSessions.Should().BeEmpty();
+            }
+        }
+
+        [Test]
+        public void UpdateSessionDuration_Should_Only_Update_Given_Active_Sessions()
+        {
+            const int twoMinutesInMilliseconds = 120 * 1000;
+
+            using (new TransactionScope())
+            {
+                // When
+                const int candidateId = 9;
+                const int sessionId = 473;
+                var startingSessions = sessionTestHelper.GetCandidateSessions(candidateId);
+
+                sessionService.UpdateSessionDuration(sessionId);
+
+                // Then
+                var newSessions = sessionTestHelper.GetCandidateSessions(candidateId).ToList();
+
+                newSessions
+                    .Where(session => session.SessionId != sessionId)
+                    .Should()
+                    .BeEquivalentTo(startingSessions.Where(session => session.SessionId != sessionId));
+
+                var newSession = newSessions.First(session => session.SessionId == sessionId);
+                DateTime.Now.Should().BeCloseTo(
+                    newSession.LoginTime.AddMinutes(newSession.Duration),

```

Comment:
Is duration in minutes? 🙂

Reply:
Yep, the duration is calculated using `DATEDIFF(minute, [LoginTime], GetUTCDate())` in `SessionService` which calculates to the nearest minute 🙂

Reply:
Okay, great. I assume it was also in minutes in the old code/query? 🙂

Reply:
That was part of the query on the HEEDLS-212 ticket description. I assume that was part of the old code but I can check if you like? 🙂

Reply:
I see, great. Yes that's from the old code so no need to check. Thanks for confirming 🙂